cmd/ci-secret-generator/main.go

```diff
@@ -195,6 +224,12 @@ func updateSecrets(config secretgenerator.Config, client secrets.Client) error {
 				errs = append(errs, errors.New(msg))
 				continue
 			}
+			if err := validateCommandOutput(out, field.Validation); err != nil {
```

Comment:
Do we have examples of validation commands we intend to use and the type of output they generate?  Could we leak secret content via the validation command.  I don't see us doing any sort of censoring here.

Reply:
Would be great if we could test this.

Reply:
We "committed a suicide" once in the past, generating the `kubeconfig` because there were deprecated error msg in the `stderr`. So we screwed up the secret which led to outage.

The card is to validate the output before updating the secrets, suggested by Bruno.

We use [this](https://github.com/openshift/release/blob/c947321a3b8e2a399ad6cc6fdc950dfea83867ce/core-services/ci-secret-generator/_config.yaml#L2) cmd to generate the token,
ideally, i would like to validate if the generated token is valid, e.g., by `oc login` with the token and `oc whoami`

Another mistake in the past: We need to verify the kubeconfig file has only one context, named as the cluster. I would like to verify that too.

We used to verify this by https://github.com/openshift/ci-tools/blob/master/images/ci-secret-generator/oc_sa_create_kubeconfig.sh

I can definitely feel some difficulty with the current implementation:
We need api-server URL to do oc-login and we have list of clusters.
How do we map the relations?
It is even more challenging to do it as "shell in yaml".

Reply:
Your concern of exposing secrets is very valid to me. (It is listed even in the notes from one of the incidents.)
I did not fix the code because I was thinking about the validation and might fix all those together. Moreover, after we fixed the config, the error does not show any more. (I know it is an excuse and does not prevent exposing from happening again. But it is the current reality).

So we have 2 issues now to me:
1. How we can censor the token? The risky is already in our code without the validation. 
    It is hard to achieve because we do not know the token in advance. We need to basically remove all the output which is not friendly at all to the debugging. (Another excuse that I did not fix it)
2. Is validation technically feasible? A typical case is the token: how should we validate it? 
    It is more functional to the card. The validation wont work easily unless we need to extend how the tool works, e.g., we need to provide api server in the validation command. I do not have the solution.